### ISSUE TITLE
Fully supervised problem

### ISSUE THREAD

**Issue by bluecat1024** [tkipf/gcn#68]
You stated that GCN reached 88% on CORA dataset when using all labels.
But when I removed the masks in metrics.py, the loss began to skyrocket in training, with accuracy extremely low at last.
What happened that the GCN model didn't act as 88% F1 score as your paper stated?

**Comment by tkipf** [comment]
Would you mind elaborating a bit more on what you mean with “removing the masks”? What would you like to test or achieve? I am not sure if I understand the question exactly. Thanks :)

**Comment by bluecat1024** [comment]
def masked_softmax_cross_entropy(preds, labels, mask):
    """Softmax cross-entropy loss with masking."""
    loss = tf.nn.softmax_cross_entropy_with_logits(logits=preds, labels=labels)
    mask = tf.cast(mask, dtype=tf.float32)
    mask /= tf.reduce_mean(mask)
    #loss *= mask
    return tf.reduce_mean(loss)


def masked_accuracy(preds, labels, mask):
    """Accuracy with masking."""
    correct_prediction = tf.equal(tf.argmax(preds, 1), tf.argmax(labels, 1))
    accuracy_all = tf.cast(correct_prediction, tf.float32)
    mask = tf.cast(mask, dtype=tf.float32)
    mask /= tf.reduce_mean(mask)
    #accuracy_all *= mask
    return tf.reduce_mean(accuracy_all)

I suppose by removing loss *= mask and that's using all labels' cross entropy
But the loss doesn't go down as it has performed as there is mask.

**Comment by tkipf** [comment]
I see now what you did there :) You're probably still using only the training labels from `y_train`, which are just zero for all elements outside of the mask, i.e. they don't carry any information for the model. If you want to train fully supervised, make sure you also provide the correct training labels.
